### PR TITLE
MGMT-7377: Remove auto-assign param and set default to SDN for network…

### PR DESCRIPTION
**Pull request by oshercc**
…twork type

# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @avishayt 
/cc @ori-amizur 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory_test.go [hunk 1]
```diff
@@ -3913,7 +3913,7 @@ var _ = Describe("cluster", func() {
 			Expect(err).ShouldNot(HaveOccurred())
 			mockClusterApi.EXPECT().VerifyClusterUpdatability(gomock.Any()).Return(nil).Times(1)
 			mockClusterApi.EXPECT().RefreshStatus(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil).Times(1)
-			networkType := "OpenShiftSDNaasd"
```
Comment (oshercc) [inline internal/bminventory/inventory_test.go:3916]: noting to see here, keep on reviewing

**Comment by avishayt** [comment]
This part looks good, but need to add the logic of choosing SDN/OVN based on IPv4/6 to the operator.

File: internal/controller/controllers/clusterdeployments_controller.go [hunk 2]
```diff
@@ -718,6 +725,35 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context,
 	return nil
 }
 
+func getClusterNetworkType(params *models.ClusterUpdateParams, cluster *common.Cluster) string {
```
Comment (filanov) [inline internal/controller/controllers/clusterdeployments_controller.go:728]: Maybe it should be called select network type?

File: internal/controller/controllers/clusterdeployments_controller.go [hunk 3]
```diff
@@ -877,6 +913,12 @@ func (r *ClusterDeploymentsReconciler) createNewCluster(
 		clusterParams.ServiceNetworkCidr = swag.String(clusterInstall.Spec.Networking.ServiceNetwork[0])
 	}
 
+	if network.IsIPv6CIDR(swag.StringValue(clusterParams.ClusterNetworkCidr)) || network.IsIPv6CIDR(swag.StringValue(clusterParams.ServiceNetworkCidr)) {
```
Comment (filanov) [inline internal/controller/controllers/clusterdeployments_controller.go:916]: if it's not required then maybe we should have this logic only update if needed?

**Review by filanov — COMMENTED** [review]
Please add unit tests

File: internal/controller/controllers/clusterdeployments_controller.go (hunk 3 above)
Comment (rollandf) [reply internal/controller/controllers/clusterdeployments_controller.go:916]: `NetworkType` is not required in creation so we can remove this part.
Update will be called immediately after.

**Review by filanov — APPROVED** [review]

**Pull request merged**